parser-typechecker/src/Unison/CommandLine/Main.hs

```diff
@@ -228,7 +228,7 @@ main dir defaultBaseLib initialPath (config,cancelConfig) initialInputs startRun
                 writeIORef pageOutput True
                 pure x) `catch` interruptHandler
       interruptHandler (asyncExceptionFromException -> Just UserInterrupt) = awaitInput
-      interruptHandler _ = pure $ Right QuitI
+      interruptHandler e = error (show e)
```

Comment:
This crashes loudly on receiving an asynchronous error, instead of silently quitting.